costs/parser.go

```diff
@@ -0,0 +1,118 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package costs
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"gopkg.in/olivere/elastic.v5"
+)
+
+// flattenSubAggregation allows to transform an array of map to one flattened map
+func flattenSubAggregation(in *interface{}) {
+	oldTab := (*in).([]interface{})
+	newMap := make(map[string]interface{})
+	for _, field := range oldTab {
+		for name, subField := range field.(map[string]interface{}) {
+			newMap[name] = subField
+		}
+	}
+	*in = newMap
+}
+
+// stripAggregationArchitecture strips an array by deleting some values and calling flattenSubAggregation
+func stripAggregationArchitecture(aggrName string, aggrType interface{}, in map[string]interface{}) {
+	key, ok := in["key_as_string"]
+	if !ok {
+		key, ok = in["key"]
+	}
+	if skey := fmt.Sprintf("%v", key); ok {
+		if aggrMapType, ok := aggrType.(map[string]interface{}); ok && aggrName[0] == '*' {
+			aggrType := aggrMapType["buckets"]
+			flattenSubAggregation(&aggrType)
+		}
+		in[skey] = aggrType
+		delete(in, "key")
+		delete(in, "key_as_string")
+	}
+	delete(in, aggrName)
+}
+
+func browseMapRecursivly(in map[string]interface{}) {
```

Comment:
`browseMapRecursively` and `browseTabRecursively` have no return value, so I expect they have side effects. Please make sure whenever functions have side effects these are clearly documented, preferably in the function name. I have no idea what sort of side effects browsing a map can have.